src/lplib/cmap.ml

```diff
@@ -0,0 +1,162 @@
+(*Type of points and intervals. Used to determine if a cursor is in the range
+of a specific token, which is an interval of points (positions)*)
+(*The presence of t and compare make RangeType an OrderedType (in the sense of Map) *)
+module type RangeType = sig
+
+  type point
+  val make_point : int -> int -> point
+
+  val line : point -> int
+  val column : point -> int
+  type cmp = Before | In | After
+
+  val point_to_string : point -> string
+
+  type t
+  type interval = t
+  (*Ensures an interval is well defined (e.g with start < finish)*)
+  val make_interval : point -> point -> interval
+
+  val interval_start : interval -> point
+  val interval_end : interval -> point
+
+  val in_range : point -> interval -> cmp
```

Comment:
thanks to move at the end of the signature (with the type cmp)

Reply:
Ok now I think.